pkg/controller/common/beat/pod.go

```diff
@@ -0,0 +1,122 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package beat
+
+import (
+	"fmt"
+	"hash"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/container"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/defaults"
+	commonhash "github.com/elastic/cloud-on-k8s/pkg/controller/common/hash"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/pointer"
+)
+
+const (
+	CAVolumeName = "es-certs"
+	CAMountPath  = "/mnt/elastic-internal/es-certs/"
+	CAFileName   = "ca.crt"
+
+	ConfigVolumeName = "config"
+	ConfigMountPath  = "/etc/beat.yml"
+	ConfigFileName   = "beat.yml"
+
+	DataVolumeName        = "data"
+	DataMountPathTemplate = "/var/lib/%s/%s/%s-data"
+	DataPathTemplate      = "/usr/share/%s/data"
+
+	// ConfigChecksumLabel is a label used to store beats config checksum.
+	ConfigChecksumLabel = "beat.k8s.elastic.co/config-checksum"
+
+	// VersionLabelName is a label used to track the version of a Beat Pod.
+	VersionLabelName = "beat.k8s.elastic.co/version"
+)
+
+func buildPodTemplate(params DriverParams, defaultImage container.Image, f func(builder *defaults.PodTemplateBuilder), checksum hash.Hash) corev1.PodTemplateSpec {
```

Comment:
I would probably have spent two more letters and called `f` `mod` or something like that. Even if it's an internal function. Just a personal preference, feel free to ignore it.